src/CoreWCF.Primitives/src/CoreWCF/Channels/ServiceChannelProxy.cs

```diff
@@ -15,7 +15,7 @@
 
 namespace CoreWCF.Channels
 {
-    internal sealed class ServiceChannelProxy : DispatchProxy, ICommunicationObject, IChannel, IClientChannel, IOutputChannel, IRequestChannel, IServiceChannel, IDuplexContextChannel
+    public class ServiceChannelProxy : DispatchProxy, ICommunicationObject, IChannel, IClientChannel, IOutputChannel, IRequestChannel, IServiceChannel, IDuplexContextChannel
```

Comment:
This change makes sense, but I'm not really happy with it. The DispatchProxy class that we use to create the proxied type needs to derive from this class via ref emit, which if why it can't be sealed and needs to be public. The thing I don't like is making this type public. Are you familiar with the `InternalsVisibleTo` attribute? [Docs are here](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-5.0). Could you see if adding an InternalsVisibleTo attribute for the DispatchProxy assembly would allow us to keep this type internal?

Reply:
I will double check this change and try to keep the class internal. Actually I just went the quick path to align it with the [System.ServiceModel.ServiceChannelProxy](https://github.com/dotnet/wcf/blob/367d64dc79a9b23acc15a3ef549996cf05e10e72/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannelProxy.cs#L18), `sealed` is definitely an issue, but maybe `internal` can even stay.

Reply:
@mconnew I fear that `public` cannot be avoided for the currently targeted runtimes unless we remove the strong name. The issue is partly related to https://github.com/dotnet/runtime/issues/30917  and some limitations with the `InternalsVisibleTo`:

I tried following things: 

- Add `[assembly: InternalsVisibleTo("ProxyBuilder")]` -> Leads to a compiler error because CoreWCF has a strong name: 

> Error CS1726 : Friend assembly reference 'ProxyBuilder' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations.

- Add `[assembly: InternalsVisibleTo("ProxyBuilder, PublicKey=null")]` or `[assembly: InternalsVisibleTo("ProxyBuilder, PublicKey=")]` or `[assembly: InternalsVisibleTo("ProxyBuilder, PublicKey=<public-key of corewcf.snk>")]`
 -> Leads to a runtime error (not always but sometimes, likely related to https://github.com/dotnet/runtime/issues/30917): 
> Type 'generatedProxy_6' from assembly 'ProxyBuilder, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' is attempting to implement an inaccessible interface.

I was the initiator for requesting strong name via #264 Things changed already for us, and nowadays we would also be fine to not have a strong name anymore. 

I think we have only two choices: 

a) Make it public
b) Remove the strong name and add a `[assembly: InternalsVisibleTo("ProxyBuilder")]`

Reply:
I fiddled a bit around with some tricks to maybe bypass some checks of the runtime but unfortunately I failed. I was not able to work around the requirement that the base class of proxies currently need to be public. 😢 

I added the Browsable, EditorBrowsable and Obsolete attribute to hide it as good as possible. Unfortunately also here we cannot activate the error flag because there is no way to supress an error level Obsolete attribute. 

I also tried to minimize the public facing API surface to zero by sealing some methods and using explicit interface implementations where possible.